### PR TITLE
Fixes #28149 - Add release documentation

### PR DESCRIPTION
**Pull request by mbacovsky**
Release Docs in Jupyter notebook format and README with instructions

### CODE REVIEW

**Comment by mbacovsky** [comment]
BTW both '.md' and '.ipynb' can be rendered by GitHub.

File: rel-eng/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+$ jupyter notebook rel-eng/gem_release.ipynb 
+```
+
+> NOTE: Every executable bit is executed in your local hammer-cli directory.
+
+> NOTE: It is recommended to use non-interactive auth with git hub (such as ssh keys)
+as current notebook can not handle password prompt. If your env requires interactive
+steps skip the step in notebook and run its equivalent manually in your terminal. 
+
+> NOTE: If you notebook hangs waiting on your prompt you have to `Menu > Kernel > Interrupt`
```
Comment (ofedoren) [inline rel-eng/README.md]: Typo?: Should it be `If your notebook`?

File: rel-eng/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Our release process is documented using Jupyter notebook (interractive documentation).
+
+To read the docs view the `gem_release.ipynb` on GitHub. To run the notebook Jupyter is needed. Jupyter is web based interractive python
+environment that can mix executable snippets with markdown.
+
+The markdown parts should inform the releaser about what is going to be done and 
+the executable pieces should do the actual work. If anything goes unexpected it is easy
+ to stop, investigate, fix and re-run the step.
+
```
Comment (ofedoren) [inline rel-eng/README.md]: Somewhere here we should also specify possible dependencies.

When I tested I hitted the following requirements:
 - transifex-client
 - python-slugify
 - asciidoc
 - To make non-interactive step 18 I needed to configure credentials for rubygems.org (I've used api key)

**Comment by mbacovsky** [comment]
@ofedoren, updated in separate commit,

**Comment by mbacovsky** [comment]
@ofedoren thanks, I have nothing to add at the moment. Hit the button! :boom:

**Comment by ofedoren** [comment]
@mbacovsky, merging then, thanks! :tada:

**Pull request merged**